### PR TITLE
Add Attachments to Broadcast

### PR DESCRIPTION
**Pull request by tanndlin**
image

### CODE REVIEW

**Comment by tanndlin** [comment]
fixes #282

File: src/app/plugins/broadcast.plugin.ts [hunk 1]
```diff
@@ -104,22 +127,26 @@ export class BroadcastPlugin extends Plugin {
...
+  private _createAnnouncement(): (RichEmbed | string[])[] {
     const embed = new RichEmbed();
-    embed.setTitle('Annoucement!');
+    embed.setTitle('Announcement!');
     embed.setColor('#ffca06');
     embed.setThumbnail(Constants.LionPFP);
-    embed.setDescription(annoucement);
+    embed.setDescription(this._ANNOUNCEMENT_CONTENT);
+
+    if (!this._ATTACHMENTS) {
```
Comment (c650) [inline src/app/plugins/broadcast.plugin.ts:143]: is empty array falsey?

File: src/app/plugins/broadcast.plugin.ts [hunk 2]
```diff
@@ -47,35 +54,51 @@ export class BroadcastPlugin extends Plugin {
...
+      `Sending Announcement to \`${this._CHANS_TO_SEND.length}\` classes... ` +
         `I will let you know it has finished`
     );
 
-    const promises = this._CHANS_TO_SEND.map((chan) =>
-      (chan as TextChannel).send(this._EMBED_TO_SEND)
-    );
-    this._EMBED_TO_SEND = null;
-    this._CHANS_TO_SEND = [];
+    const [announcementEmbed, attachmentEmbed] = embeds;
```
Comment (c650) [inline src/app/plugins/broadcast.plugin.ts:91]: is it actually an attachment *embed*

File: src/app/plugins/broadcast.plugin.ts [hunk 3]
```diff
@@ -88,11 +88,11 @@ export class BroadcastPlugin extends Plugin {
...
     );
 
-    const [announcementEmbed, attachmentEmbed] = embeds;
+    const [announcementEmbed, attachments] = embeds;
     for (const chan of this._CHANS_TO_SEND) {
       //Send announcement first, then the attachments
       await (chan as TextChannel).send(announcementEmbed);
-      await (chan as TextChannel).send(attachmentEmbed);
+      await (chan as TextChannel).send(attachments);
     }
```
Comment (c650) [inline src/app/plugins/broadcast.plugin.ts]: why not promise.all it

**Review by c650 — APPROVED** [review]

File: src/app/plugins/broadcast.plugin.ts (hunk 3 above)
Comment (tanndlin) [reply src/app/plugins/broadcast.plugin.ts]: To ensure the embed comes first

Comment (c650) [reply src/app/plugins/broadcast.plugin.ts]:
```js

await Promise.all( this._CHANS_TO_SEND.map( async (c) => {
  await  c.send(announcement);
  await c.sense(Attachments);
});

```

**Review by c650 — APPROVED** [review]

**Pull request merged**